crates/sui-types/src/base_types.rs

```diff
@@ -94,20 +105,178 @@ pub fn random_object_ref() -> ObjectRef {
     )
 }
 
+/// Wrapper around StructTag with a space-efficient representation for common types like coins
+/// The StructTag for a gas coin is 84 bytes, so using 1 byte instead is a win
+#[derive(Eq, PartialEq, PartialOrd, Ord, Debug, Clone, Deserialize, Serialize, Hash)]
+pub enum MoveType {
```

Comment:
mega nit
```suggestion
pub enum MoveObjectType {
```

Reply:
My reasoning is that this is our special type for `MoveObject`, not somehow a copy of `TypeTag` or something similar.